_rules/lists-follow-content-model-a73be2.md

```diff
@@ -0,0 +1,277 @@
+<ul>
+	<li>Eggs</li>
+	<li>Milk</li>
+</ul>
+```
+
+#### Passed Example 2
+
+This `dl` element has `dt` and `dd` [children][child] which are allowed by its [content model][].
+
+```html
+<dl>
+	<dt>Physics</dt>
+	<dd>8/10</dd>
+</dl>
+```
+
+#### Passed Example 3
+
+This `ul` element has no [child][] [elements][element]. Its only [children][child] are a comment and an empty text node, neither of which are [included in the accessibility tree][].
+
+```html
+<ul>
+	<!-- empty list -->
+</ul>
+```
+
+#### Passed Example 4
+
+This `menu` element has `li` [children][child] which are allowed by its [content model][].
+
+```html
+<menu>
+	<li>New file</li>
+	<li>Save file</li>
+</menu>
+```
+
+#### Passed Example 5
+
+This `ol` element has `span` [children][child] with an [explicit role][] or `listitem`. The [content model][] of `ol` allows `li` elements which have [implicit ARIA semantics][] of `listitem`.
+
+```html
+<ol>
+	<span role="listitem">Eggs</span>
+	<span role="listitem">Milk</span>
+</ol>
+```
+
+#### Passed Example 6
+
+This `ul` element has `li` and `template` [children][child]. `li` is explicitly allowed by the `ul` [content model][], `template` is a script-supported element, which is also allowed by the `ul` content model.
+
+```html
+<ol>
+	<li>Eggs</li>
+	<li>Milk</li>
+	<template>
+		<li>{{item}}</li>
+	</template>
+</ol>
+```
+
+#### Passed Example 7
+
+This `dl` element has `div` [child][child] which is allowed by its [content model][]. The `div` element has a `dt` [child][] followed by a `dd` element, which is allowed by the `div` [content model][].
+
+```html
+<dl>
+	<div>
+		<dt>Physics</dt>
+		<dd>8/10</dd>
+	</div>
+</dl>
+```
+
+#### Passed Example 8
+
+This `dl` element has `div` [children][child] with an [explicit role][] of `term` and `definition`. The [content model][] of `dl` allows `dt` and `dd` elements which have [implicit ARIA semantics][] of `term` and `definition`.
+
+```html
+<dl>
+	<div role="term">Physics</div>
+	<div role="definition">8/10</div>
+</dl>
+```
+
+#### Passed Example 9
+
+This `dl` element has `div` [child][child] which is allowed by its [content model][]. The `div` element has `div` [children][child] with an [explicit role][] of `term` and `definition`. The [content model][] of `div` allows `dt` and `dd` elements which have [implicit ARIA semantics][] of `term` and `definition`.
+
+```html
+<dl>
+	<div>
+		<div role="term">Chemistry</div>
+		<div role="definition">7/10</div>
+	</div>
+</dl>
+```
+
+#### Passed Example 10
+
+This `ul` element has a `p` [child][] which is not [included in the accessibility tree][] because of the `hidden` attribute.
+
+```html
+<ul>
+	<p hidden>Empty list</p>
+</ul>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `ul` element had a text node as a child, which will be [included in the accessibility tree][]. Except if they only have [ascii whitespace][], text nodes are not allowed by the [content model][] of `ul` elements.
+
+```html
+<ol>
+	Coming soon!
+</ol>
```

Comment:
I'm actually not sure this is failing the SC. There is no structure conveyed through presentation, so I don't think 1.3.1 applies.
(this is not about having only one "item", this is about nothing makes it look like an item)

Reply:
The issue with this is that this content can't be predictably styled. If someone wants to make sure all list items have the same bullet. The could style it with something like `ul:not([role]) > li, ul:not([role]) > [role^=listitem]`. That won't work for this, even though they might still have styles that apply to the `ul` element.

This SC doesn't just mean that content that look a certain way can be understood that way; the inverse is true too. Even if you mark every heading up with proper heading markup, if you mark non-headings up with heading markup, there's still an issue. Same logic applies here. This isn't a list. It shouldn't be marked up as a list otherwise it makes it impossible to properly style actual lists.

Reply:
@ShadowBB agreed with me on this BTW.

Reply:
I do not fully follow the argument.

> The issue with this is that this content can't be predictably styled. 

The SC is not about content being stylable.

> This SC doesn't just mean that content that look a certain way can be understood that way; the inverse is true too. Even if you mark every heading up with proper heading markup, if you mark non-headings up with heading markup, there's still an issue. Same logic applies here. This isn't a list. It shouldn't be marked up as a list otherwise it makes it impossible to properly style actual lists.

The SC is pretty one-sided: "Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text." so I am not sure what you refer too when saying that "the inverse is true too".

Notably, your example with heading seems to imply that you consider `<div role="heading">` as a failure of 1.3.1.

This example (`<ul>Foo</ul>`) has essentially no presentation (just some indentation, looks similar to a `<div style="text-indent: 40px">Foo</div>`). I say that since no structure is conveyed through presentation, 1.3.1 doesn't care.
I understand your argument as "presentation conveys the information that this is not a list, so accessibility tree must conveys the information that this is not a list". Please correct me if I'm misunderstanding.

If that interpretation is correct, it means for example that `<div role="list"><span role="listitem">Foo</span></div>` is failing 1.3.1 (here also, there is no presentation, so presentation conveys the information that this is not a list. Yet this is [Passed Example 1 of ARIA required owned elements](https://act-rules.github.io/rules/bc4a75#passed-example-1). Which leaves me fairly confused.

---
All this might get clarified when (if) we merge this rule with the "ARIA required owned elements"…